### PR TITLE
Remove ZooKeeper support - Part I

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Task

### Description

This PR starts the removal of the ZooKeeper support from Strimzi. This initial part does the following:
* Adds protection against damaging ZooKeeper-based clusters and operates only KRaft clusters
* Removes most of ZooKeeper and migration from the `cluster-operator` module
* Removes ZooKeeper and migration from the `kafka-agent` module
* Removes ZooKeepers from the container image scripts
* Disables the ZooKeeper-based upgrade tests

The focus of this PR is on removing the ZooKeeper & KRaft migration logic and the related dead code. The only really new code is the validation of the clusters that they are KRaft-based and throwing exceptions if not in the `KafkaAssemblyOperator` class (and the related unit tests).

This PR does not handle examples, `api` module, documentation, system tests and other parts of the code. That will be handled only in separate follow-up PRs.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Update CHANGELOG.md

### CODE REVIEW

**Review by im-konge — APPROVED** [review]
LGTM, thanks :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,8 +2,13 @@
 
 ## 0.46.0
 
+* Support for ZooKeeper-based Apache Kafka clusters and for KRaft migration has been removed
+* Support for MirrorMaker 1 has been removed
+
 ### Major changes, deprecations and removals
 
+* **Support for ZooKeeper-based clusters and for migration from ZooKeeper-based clusters to KRaft has been remove.**
```
Comment (ppatierno) [inline CHANGELOG.md:10]:
```suggestion
* **Support for ZooKeeper-based clusters and for migration from ZooKeeper-based clusters to KRaft has been removed.**
```

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java [hunk 2]
```diff
@@ -1155,47 +1143,8 @@ protected Future<Void> addListenersToKafkaStatus(KafkaStatus kafkaStatus) {
...
-                            ));
-                    zkMigrationStateChecked = true;
-                    break;
-                } catch (RuntimeException e) {
-                    LOGGER.debugCr(reconciliation, "Error on checking ZooKeeper migration state on controller {}", controller.podName());
-                }
-            }
-            return zkMigrationStateChecked ? Future.succeededFuture() : Future.failedFuture(new Throwable("Failed to check ZooKeeper migration state"));
-        }
+        // TODO: Do we need this as a separate method?
```
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java:1146]: don't think it's needed.

**Review by ppatierno — APPROVED** [review]
LGTM. Assuming all tests pass.

**Pull request merged**